src/flint/test/test.py

```diff
@@ -1813,6 +1830,324 @@ def test_fmpz_mod_dlog():
             x = g.discrete_log(a)
             assert g**x == a
 
+def test_fmpz_mod_poly():
+    from flint import fmpz_poly, fmpz_mod_poly, fmpz_mod_poly_ctx, fmpz_mod_ctx, fmpz
+
+    # fmpz_mod_poly_ctx tests
+    F = fmpz_mod_ctx(11)
+    R1 = fmpz_mod_poly_ctx(F)
+    R2 = fmpz_mod_poly_ctx(11)
+    R3 = fmpz_mod_poly_ctx(13)
+
+    assert raises(lambda: fmpz_mod_ctx("AAA"), TypeError)
+    assert raises(lambda: fmpz_mod_ctx(-1), ValueError)
+    assert (R1 == R1) is True
+    assert (R1 == R2) is True
+    assert (R1 != R3) is True
+    assert (R1 != "AAA") is True
```

Comment:
We should also have tests like
```python
assert (R1 != R1) is False
assert (R1 == R3) is False
...
```
Any time you want to test a function that returns True or False you at least need one test for each case. Otherwise it is trivially easy to satisfy the tests:
```python
def __eq__(f, g):
    return True
```